### DISCUSSION TITLE
how to use MaxLength Plugin

### DISCUSSION THREAD

**Discussion by duc-gp** [facebook/lexical#4809] [Q&A]
hello,

ive seen that a MaxLengthPlugin has been added here https://github.com/facebook/lexical/pull/2254 but in my project I cannot import it. the package json states following dependencies:
"lexical": "^0.11.3",
"@lexical/react": "^0.11.3",

what I'm I doing wrong?

thanks and best

**Comment by PrasKams** [comment]
Hey- not sure if you got around your problem, but i got it working by creating custom plugin based on 

https://github.com/facebook/lexical/blob/main/packages/lexical-playground/src/plugins/MaxLengthPlugin/index.tsx

and passing the maxLength as prop to the rich text editor
`<MaxInputLengthPlugin maxLength = {maxInput} />`
My `<MaxInputLengthPlugin />` custom plugin has the same code as the git link above
I believe MaxLengthPlugin is not added to `@Lexical/react/....plugins`

**Comment by sinujohn91** [comment]
I got it working.

My deps are:
```json
"@lexical/react": "^0.33.1"
"lexical": "^0.33.1",
```

imports:
```js
import { OverflowNode } from '@lexical/overflow';
import { CharacterLimitPlugin } from '@lexical/react/LexicalCharacterLimitPlugin';
```
`initialConfig` to be passed to the `LexicalComposer`
```js
let initialConfig = {
        namespace: 'Editor',
        theme,
        onError,
        nodes: [OverflowNode],
        editable: !disabled,
 };
```
Add the plugin inside `LexicalComposer `
```js
 <CharacterLimitPlugin maxLength={maxLength} />
```
You can also pass a custom renderer. You can refer for the usage here: https://github.com/facebook/lexical/blob/main/packages/lexical-react/src/LexicalCharacterLimitPlugin.tsx#L55
